net-im/telegram-desktop/telegram-desktop-4.10.3.ebuild

```diff
@@ -0,0 +1,231 @@
+# Copyright 2020-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{9..12} )
+
+inherit xdg cmake python-any-r1 optfeature flag-o-matic
+
+DESCRIPTION="Official desktop client for Telegram"
+HOMEPAGE="https://desktop.telegram.org"
+
+MY_P="tdesktop-${PV}-full"
+SRC_URI="https://github.com/telegramdesktop/tdesktop/releases/download/v${PV}/${MY_P}.tar.gz"
+S="${WORKDIR}/${MY_P}"
+
+LICENSE="BSD GPL-3-with-openssl-exception LGPL-2+"
+SLOT="0"
+KEYWORDS="~amd64 ~riscv"
+IUSE="dbus enchant +fonts +jemalloc screencast qt6 qt6-imageformats wayland webkit +X"
+REQUIRED_USE="
+	qt6-imageformats? ( qt6 )
+	webkit? ( wayland )
+"
+# webkit -> see Telegram/lib_webview/CMakeLists.txt
+
+KIMAGEFORMATS_RDEPEND="
+	media-libs/libavif:=
+	media-libs/libheif:=
+	media-libs/libjxl
+"
+CDEPEND="
+	!net-im/telegram-desktop-bin
+	app-arch/lz4:=
+	dev-cpp/abseil-cpp:=
+	>=dev-cpp/glibmm-2.77:2.68
+	dev-libs/glib:2
+	dev-libs/libdispatch
+	dev-libs/libsigc++:2
+	dev-libs/openssl:=
+	dev-libs/protobuf
+	dev-libs/xxhash
+	media-libs/fontconfig:=
+	media-libs/libjpeg-turbo:=
+	~media-libs/libtgvoip-2.4.4_p20221208
+	media-libs/openal
+	media-libs/opus
+	media-libs/rnnoise
+	~media-libs/tg_owt-0_pre20230921:=[screencast=,X=]
+	media-video/ffmpeg:=[opus,vpx]
+	sys-libs/zlib:=[minizip]
+	virtual/opengl
+	!enchant? ( >=app-text/hunspell-1.7:= )
+	enchant? ( app-text/enchant:= )
+	jemalloc? ( dev-libs/jemalloc:=[-lazy-lock] )
+	!qt6? (
+		>=dev-qt/qtcore-5.15:5=
+		>=dev-qt/qtgui-5.15:5[dbus=,jpeg,png,wayland?,X?]
+		>=dev-qt/qtimageformats-5.15:5
+		>=dev-qt/qtnetwork-5.15:5[ssl]
+		>=dev-qt/qtsvg-5.15:5
+		>=dev-qt/qtwidgets-5.15:5[png,X?]
+		kde-frameworks/kcoreaddons:5
+		webkit? ( >=dev-qt/qtdeclarative-5.15:5 )
+	)
+	qt6? (
+		>=dev-qt/qtbase-6.5:6[dbus=,gui,network,opengl,wayland?,widgets,X?]=
```

Comment:
Drop the = at end to fix pkgcheck, should update metadata.xml with current telegram useflags as well.

Reply:
The `=` is required (and is a valid operator as far as I'm aware). You cannot build build telegram without Qt6DBus when dev-qt/qtbase was built with USE=dbus, as Qt6Gui will depend on Qt6DBus then.

Reply:
In qt's own packages, sometime had to resort to sed instead given disabling sub-component of qtbase tend to end badly.

Reply:
@joecool1029 Oh right, I always get the position of the `=` wrong. It should be before the `[`

@ionenwks You're right. I should rethink how I control these dependencies. I don't want to force wayland but I don't like having to `sed` my way around everything or have bespoke patches...

Reply:
The others are pretty safe, aka wayland's diable_find_package which is not part of qtbase is fine. It's when you disable bits of qtbase (Gui, Widgets, DBus, Opengl, etc...) that you end up breaking assumptions. So it'd be more an exception that a need to change everything. 

On a side-note, disabling Quick will force disable QuickWidgets in a similar way (aka QuickWidgets tries to find Quick, then fails). So specifying both is not strictly needed.

Reply:
I prefer both as it helps me keep track of the automagic deps I've handled :)
Thanks a lot for the comment, I guess I can handle the DBus dependency through `sed` and the rest through find_package disabling.